.github/workflows/build-publish-containers.yml

```diff
@@ -58,71 +57,19 @@ jobs:
       - name: Convert and store repository owner in lowercase
         run: |
           echo REPOSITORY_OWNER=$(echo ${{ github.repository_owner }} | tr '[:upper:]' '[:lower:]') >> $GITHUB_ENV
-          echo DOCKERFILE=${{ matrix.tag }}/Dockerfile >> $GITHUB_ENV
+          echo IMAGE_TAG=$(date +"%F")-{{ github.GITHUB_RUN_ID }} >> $GITHUB_ENV
 
       - name: Build and push to GitHub Packages
         uses: docker/build-push-action@v3
         with:
-          tags: ghcr.io/${{ env.REPOSITORY_OWNER }}/${{ matrix.tag }}-${{ matrix.platform }}
-          file: ${{ env.DOCKERFILE }}
-          platforms: linux/${{ matrix.platform }}
+          tags: ghcr.io/${{ env.REPOSITORY_OWNER }}/${{ matrix.tag }}:{{ env.IMAGE_TAG }}
+          context: ${{ matrix.tag }}
+          platforms: linux/amd64,linux/arm64
```

Comment:
That's the major shift right here. By not creating a platform+tag matrix, you build everything in one step for each tag.